spec.html

```diff
@@ -42237,89 +42232,77 @@ <h1>%TypedArray%.prototype.set ( _source_ [ , _offset_ ] )</h1>
         </emu-alg>
         <p>This method is not generic. The *this* value must be an object with a [[TypedArrayName]] internal slot.</p>
 
-        <emu-clause id="sec-settypedarrayfromtypedarray" type="abstract operation" oldids="sec-%typedarray%.prototype.set-typedarray-offset">
+        <emu-clause id="sec-settypedarrayfromarraylike" type="abstract operation" oldids="sec-%typedarray%.prototype.set-array-offset">
           <h1>
-            SetTypedArrayFromTypedArray (
+            SetTypedArrayFromArrayLike (
               _target_: a TypedArray,
               _targetOffset_: a non-negative integer or +&infin;,
-              _source_: a TypedArray,
+              _source_: an ECMAScript language value, but not a TypedArray,
             ): either a normal completion containing ~unused~ or a throw completion
           </h1>
           <dl class="header">
             <dt>description</dt>
             <dd>It sets multiple values in _target_, starting at index _targetOffset_, reading the values from _source_.</dd>
           </dl>
           <emu-alg>
-            1. Let _targetBuffer_ be _target_.[[ViewedArrayBuffer]].
-            1. Let _targetRecord_ be MakeTypedArrayWithBufferWitnessRecord(_target_, ~seq-cst~).
-            1. If IsTypedArrayOutOfBounds(_targetRecord_) is *true*, throw a *TypeError* exception.
-            1. Let _targetLength_ be TypedArrayLength(_targetRecord_).
-            1. Let _srcBuffer_ be _source_.[[ViewedArrayBuffer]].
-            1. Let _srcRecord_ be MakeTypedArrayWithBufferWitnessRecord(_source_, ~seq-cst~).
-            1. If IsTypedArrayOutOfBounds(_srcRecord_) is *true*, throw a *TypeError* exception.
-            1. Let _srcLength_ be TypedArrayLength(_srcRecord_).
-            1. Let _targetType_ be TypedArrayElementType(_target_).
-            1. Let _targetElementSize_ be TypedArrayElementSize(_target_).
-            1. Let _targetByteOffset_ be _target_.[[ByteOffset]].
-            1. Let _srcType_ be TypedArrayElementType(_source_).
-            1. Let _srcElementSize_ be TypedArrayElementSize(_source_).
-            1. Let _srcByteOffset_ be _source_.[[ByteOffset]].
-            1. If _targetOffset_ = +∞, throw a *RangeError* exception.
```

Comment:
I believe we intentionally separated out these comparisons to +∞ to avoid doing arithmetic on infinities, though it appears we *do* define it in §5.2.5 Mathematical Operations, so I guess it's fine. But I'd want the other editors to weigh in first.